### DISCUSSION TITLE
Converting Car Position Data to GPS Coordinates

### DISCUSSION THREAD

**Discussion by ac1987** [theOehrly/Fast-F1#436] [General]
Hello,
I am attempting to convert a car’s position throughout a lap to a GPX file, in order to overlay car position on a satellite map. See an attempt below.

Currently, I convert the X and Y position data to decimal latitude and longitude and add them to the GPS coordinates of the relevant track provided by an ergast call. 

However, it appears the origin provided by ergast and the origin used for the car’s position are different. You can see in the image below the trace is offset slightly southwest from the track. 

Is there a way, or an alternative source (other than ergast), to get the GPS coordinates representing the origin of the car’s position data to more accurately overlay the car’s path to the track? 

Circuit of the Americas, 2022 Qualifying, SAI’s fastest lap.
![image](https://github.com/theOehrly/Fast-F1/assets/5466161/a8fe5a84-39d8-49c3-b8de-7f62a381ee0a)

**Comment by theOehrly** [comment]
I don't know how the origin for the car position data is chosen. Maybe it doesn't even have any particular meaning and F1 simply chooses whatever is convenient. 
Additionally, I don't know how Ergast decides on the coordinates for the track. 

I'm pretty sure you won't find GPS coordinates for the origin of the car position. In the end, this is just some internal thing on the F1 API. I fear that you'll need to figure those out manually for each track.
